internal/configs/hcl2shim/mock_value_composer.go

```diff
@@ -96,15 +86,24 @@ func (mvc MockValueComposer) composeMockValueForAttributes(schema *configschema.
 		// If the value present in configuration - just use it.
 		if cv, ok := configMap[k]; ok && !cv.IsNull() {
 			mockAttrs[k] = cv
-			addPotentialDefaultsWarning(k, "The field is ignored since overriding configuration values is not allowed.")
+			diags = diags.Append(tfdiags.WholeContainingBody(
+				tfdiags.Error,
+				fmt.Sprintf("Invalid mock/override field `%v`", k),
+				fmt.Sprintf(k, "The field is ignored since overriding configuration values is not allowed."),
+			))
```

Comment:
We need to check if the value exists in `defaults` map, otherwise it is not an error.